src/modules/job-list/job_data.c

```diff
@@ -117,6 +118,30 @@ static const char *parse_job_name (const char *path)
     return path;
 }
 
+static int parse_per_resource (struct job *job,
+                               const char **type,
+                               int *count)
+{
+    json_error_t error;
+
+    if (json_unpack_ex (job->jobspec, &error, 0,
+                        "{s:{s:{s?:{s?:{s?:{s?:s s?:i}}}}}}",
+                        "attributes",
+                          "system",
+                            "shell",
+                              "options",
+                                "per-resource",
+                                  "type", type,
+                                  "count",count) < 0) {
+        flux_log (job->h, LOG_ERR,
+                  "%s: job %ju invalid jobspec: %s",
```

Comment:
See `get_per_resource_option()` in the shell info.c - per-resource must contain a type key or it's invalid.